jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/StepsRunner.java

```diff
@@ -430,89 +434,138 @@ private void checkImageInTargetRegistry() {
 
     results.manifestCheckResult =
         executorService.submit(
-            () ->
-                new CheckImageStep(
-                        buildContext,
-                        childProgressDispatcherFactory,
-                        results.targetRegistryClient.get(),
-                        Verify.verifyNotNull(
-                                results
-                                    .builtImagesAndContainerConfigurationPushResults
-                                    .get()
-                                    .get(results.builtImages.get().get(0)))
-                            .get(),
-                        results.builtImages.get().get(0).get())
-                    .call());
+            () -> {
+              Map.Entry<Future<Image>, Image> entry =
+                  results.builtImagesAndBaseImages.get().entrySet().iterator().next();
+              Future<Image> builtImage = entry.getKey();
```

Comment:
You only need the key, so no need for `entrySet()`. And let's add a check; I'm certain thecheck will become useful going forward.
```suggestion
              Verify.verify(results.builtImagesAndBaseImages.get().size() == 1);
              Future<Image> builtImage = results.builtImagesAndBaseImages.get().keySet().iterator().next();
```